### PR TITLE
Move methods from ConversationListViewController to ZClientViewController

### PR DESCRIPTION
**Pull request by billypchan**
### Issues
The methods `showDataUsagePermissionDial` and `showAvailabilityBehaviourChangeAlertIfNeeded` in
`ConversationListViewController` are not related to the class

### Solutions

Put the methods in `ZClientViewController` and clean up unnecessary properties `isComingFromRegistration` & `needToShowDataUsagePermissionDialog`.
Refactor viewDidAppear to Swift.

### CODE REVIEW

File: Wire-iOS Tests/ZClientViewControllerSnapshotTests.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    override func tearDown() {
+        sut = nil
+        coreDataFixture = nil
+        
+        super.tearDown()
+    }
+
+    func testForShowDataUsagePermissionDialogIfNeeded() {
```
Comment (typfel) [inline Wire-iOS Tests/ZClientViewControllerSnapshotTests.swift]: I don't think this test doesn't needs to be a snapshot test. This seems like the perfect opportunity to for a unit test on a view model.

File: Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift [hunk 2]
```diff
@@ -21,6 +21,16 @@ import Foundation
...
 extension ZClientViewController {
 
+    // MARK: - Setup methods
+
+    @objc
+    func setupConversationListViewController(account: Account) {
+        conversationListViewController = ConversationListViewController()
+        conversationListViewController.account = account
+
+        needToShowDataUsagePermissionDialog = false
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift]: This seems unrelated to this method can't it be moved to the `init` just like `self.dataUsagePermissionDialogDisplayed = false`

File: Wire-iOS Tests/ZClientViewControllerSnapshotTests.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS Tests/ZClientViewControllerSnapshotTests.swift]: refactored to test the alert is created or not for the different stages.

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+PushPermissions.swift [hunk 3]
```diff
@@ -52,7 +52,12 @@ extension ConversationListViewController {
         contentContainer.alpha = 1.0
     }
 
-    @objc func showPushPermissionDeniedDialogIfNeeded() {
+    private var isComingFromRegistration: Bool {
+        return ZClientViewController.shared()?.isComingFromRegistration ?? false
+    }
+
+    @objc
+    func showPushPermissionDeniedDialogIfNeeded() {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+PushPermissions.swift]: I guess this dialogue is also not directly related to the conversation list. Can of course be re-factored in a separate PR.

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+PushPermissions.swift]: This does not add a alert but a `PermissionDeniedViewController` to `ConversationListViewController` so it should stays here.
